src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -341,51 +345,67 @@ static void read_channel_args(grpc_chttp2_transport* t,
     int min;
     int max;
     bool availability[2] /* server, client */;
-  } settings_map[] = {{GRPC_ARG_MAX_CONCURRENT_STREAMS,
-                       GRPC_CHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS,
-                       -1,
-                       0,
-                       INT32_MAX,
-                       {true, false}},
-                      {GRPC_ARG_HTTP2_HPACK_TABLE_SIZE_DECODER,
-                       GRPC_CHTTP2_SETTINGS_HEADER_TABLE_SIZE,
-                       -1,
-                       0,
-                       INT32_MAX,
-                       {true, true}},
-                      {GRPC_ARG_MAX_METADATA_SIZE,
-                       GRPC_CHTTP2_SETTINGS_MAX_HEADER_LIST_SIZE,
-                       -1,
-                       0,
-                       INT32_MAX,
-                       {true, true}},
-                      {GRPC_ARG_HTTP2_MAX_FRAME_SIZE,
-                       GRPC_CHTTP2_SETTINGS_MAX_FRAME_SIZE,
-                       -1,
-                       16384,
-                       16777215,
-                       {true, true}},
-                      {GRPC_ARG_HTTP2_ENABLE_TRUE_BINARY,
-                       GRPC_CHTTP2_SETTINGS_GRPC_ALLOW_TRUE_BINARY_METADATA,
-                       1,
-                       0,
-                       1,
-                       {true, true}},
-                      {GRPC_ARG_HTTP2_STREAM_LOOKAHEAD_BYTES,
-                       GRPC_CHTTP2_SETTINGS_INITIAL_WINDOW_SIZE,
-                       -1,
-                       5,
-                       INT32_MAX,
-                       {true, true}}};
+  } settings_map[] = {
+      {GRPC_ARG_MAX_CONCURRENT_STREAMS,
+       GRPC_CHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS,
+       -1,
+       0,
+       INT32_MAX,
+       {true, false}},
+      {GRPC_ARG_HTTP2_HPACK_TABLE_SIZE_DECODER,
+       GRPC_CHTTP2_SETTINGS_HEADER_TABLE_SIZE,
+       -1,
+       0,
+       INT32_MAX,
+       {true, true}},
+      {GRPC_ARG_MAX_METADATA_SIZE,
+       GRPC_CHTTP2_SETTINGS_MAX_HEADER_LIST_SIZE,
+       -1,
+       0,
+       INT32_MAX,
+       {true, true}},
+      {GRPC_ARG_HTTP2_MAX_FRAME_SIZE,
+       GRPC_CHTTP2_SETTINGS_MAX_FRAME_SIZE,
+       -1,
+       16384,
+       16777215,
+       {true, true}},
+      {GRPC_ARG_HTTP2_ENABLE_TRUE_BINARY,
+       GRPC_CHTTP2_SETTINGS_GRPC_ALLOW_TRUE_BINARY_METADATA,
+       1,
+       0,
+       1,
+       {true, true}},
+      {GRPC_ARG_HTTP2_STREAM_LOOKAHEAD_BYTES,
+       GRPC_CHTTP2_SETTINGS_INITIAL_WINDOW_SIZE,
+       -1,
+       5,
+       INT32_MAX,
+       {true, true}},
+      {GRPC_ARG_EXPERIMENTAL_HTTP2_PREFERRED_CRYPTO_FRAME_SIZE,
```

Comment:
I'm gonna be a pain...

I think this shouldn't be `GRPC_ARG_EXPERIMENTAL_HTTP2_PREFERRED_CRYPTO_FRAME_SIZE` but instead a flag that allows sending this setting... `grpc.experimental.http2_advertise_preferred_crypto_frame_size` maybe? (and so we'll need a different parsing method)

We don't really want users tuning this, instead we want to be able to tune it for them based on memory pressure.

Reply:
I made some updates:

* converted channel arg GRPC_ARG_EXPERIMENTAL_HTTP2_PREFERRED_CRYPTO_FRAME_SIZE to bool. If set to true, it sends a preferred crypto frame size value of INT_MAX at connection startup.
* I think the peer_state_based_framing experiment is no longer required so I removed its uses in the chttp2_transport.cc file. I didn't remove the experiment itself but I think that can be done in the next PR. I updated the comments in the code so hopefully that would explain why I think this experiment is no longer needed.